ffi/framebuffer_pocketbook.lua

```diff
@@ -16,22 +16,45 @@ local function _getPhysicalRect(fb, x, y, w, h)
     return bb:getPhysicalRect(x, y, w, h)
 end
 
-local function _updatePartial(fb, x, y, w, h)
+local function _adjustAreaColours(fb)
+    if fb.device.hasColorScreen and fb.device.color_saturation then
```

Comment:
The `dither` shenanigans means you probably don't even need this pair of checks anymore.

Reply:
Totally agree, the reason I kept them is when hw support for dithering would be implemented in a pocketbook later on with a non-color screen this code will cause breakage for that implementation if those checks are gone.

This way it's also easier to understand that you need a color screen and a saturation setting to make this code work properly.